SS14.Shared/GameObjects/Components/Container/Container.cs

```
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using SS14.Shared.GameObjects;
+using SS14.Shared.Interfaces.GameObjects;
+using SS14.Server.Interfaces.GameObjects;
+using SS14.Shared.Interfaces.GameObjects.Components;
+using SS14.Shared.IoC;
+
+namespace SS14.Server.GameObjects.Components.Container
+{
+    public class Container : List<IEntity>, IContainer
+    {
+        public Container(IComponent holder)
+        {
+            IContainerManager containermanager = holder.Owner.GetComponent<IContainerManager>();
+            if (containermanager == null)
+            {
+                var factory = IoCManager.Resolve<IComponentFactory>();
+                holder.Owner.AddComponent(factory.GetComponent<ContainerManagerComponent>());
+            }
+            Owner = holder.Owner;
+            if(containermanager.EntityContainers[holder.GetType()] != null)
+            {
+                throw new NotImplementedException(); //This system is designed to have singular containers per component, make a new component dummy.
+            }
+            containermanager.EntityContainers[holder.GetType()] = this;
+        }
+
+        private IEntity Owner { get; set; }
+
+        public bool CanInsert(IEntity toinsert)
+        {
+            if (toinsert.GetComponent<ITransformComponent>().ContainsEntity(Owner.GetComponent<ITransformComponent>())) //Crucial, prevent circular insertion
+            {
+                return false;
+            }
+            return true;
+        }
+
+        public bool Insert(IEntity toinsert)
```

Ok I'm fine with keeping this boolean return, because user code can block it, but I do still think that the infinite loop prevention should throw an exception.